plugins/org.jkiss.dbeaver.model.sql/src/org/jkiss/dbeaver/model/sql/parser/SQLParserContext.java

```diff
@@ -74,7 +103,24 @@ public IDocument getDocument() {
     }
 
     public SQLDialect getDialect() {
```

Comment:
I still don't like that the method with such a complex thing as dialect instantiation is called like just a getter

Reply:
Probably it should be part of refactoring, but I'm not sure that it related to current fix, isn't?

Reply:
You change the whole logic of this method. Since you're doing it, it's a part of this PR